### DISCUSSION TITLE
Compile Shared and Static Libraries in addition to Executables

### DISCUSSION THREAD

**Discussion by mikeseese** [denoland/deno#9245] [Ideas]
I'm interfacing with a 3rd party system that allows plugins through dynamic/shared libraries. That system has a tried and true build system using C++ to build those libraries, but it would be really interesting/powerful to be able to also build those plugins with JS/TS. I noticed that Deno recently released with the unstable Compile flag to compile to an executable, but I was curious if it's even something to consider to expand that to libraries that can work with other systems.

Any thoughts on something like this?

**Comment by mikeseese** [comment]
Any thoughts @lucacasonato (since you did the PR for the initial category)?

**Comment by mikeseese** [comment]
FWIW, I went through the code again with fresh eyes, and this doesn't seem to be a logically easy pivot from the existing compile functionality. The existing functionality just appends the source to the binary and extracts and runs it later. This is likely a non-trivial feature.

**Comment by FGasper** [comment]
If I understand this feature request correctly, it would allow embedding Deno inside a C/C++ application, or linking to Deno as a shared library?

That would be _amazing_!

**Comment by mikeseese** [comment]
Yes, that is the intent of this request

**Comment by phenax** [comment]
This would be awesome to have! Is anyone looking into this?
